### PR TITLE
Add support for push events

### PR DESCRIPTION
**Pull request by williamdes**
- Updated README
- Added composer.json
- Added gihub_webhook_push in lib/github.php
- Updated gitignore
- Added push.php

Fixes: https://github.com/phpmyadmin/phpmyadmin/issues/14287

Supports
- sending via SMTP server
- sending via `mail()` function of php
- custom headers

### CODE REVIEW

File: hooks/push.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    $mail->SMTPDebug  = 0;
+    $mail->isSMTP();
+    $mail->Host       = SMTP_HOST;
+    $mail->SMTPAuth   = true;
+    $mail->Username   = SMTP_USERNAME;
+    $mail->Password   = SMTP_PASSWORD;
+    $mail->SMTPSecure = SMTP_MODE;
+    $mail->Port       = SMTP_PORT;
+
+    $mail->setFrom(SMTP_SEND_FROM, $data->authorName);
```
Comment (williamdes) [inline hooks/push.php]: Replacing `SMTP_SEND_FROM` with `$data->authorEmail` to match what exists, Is it a good idea ?

Comment (nijel) [reply hooks/push.php]: AFAIK this is how current hook works.

File: hooks/push.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+try {
+
+    $mail->SMTPDebug  = 0;
+    $mail->isSMTP();
+    $mail->Host       = SMTP_HOST;
+    $mail->SMTPAuth   = true;
+    $mail->Username   = SMTP_USERNAME;
+    $mail->Password   = SMTP_PASSWORD;
+    $mail->SMTPSecure = SMTP_MODE;
+    $mail->Port       = SMTP_PORT;
```
Comment (nijel) [inline hooks/push.php]: Can you please also support sendmail here? That would make it much easier to configure on existing infrastructure - sendmail is already working there, while we don't have remote SMTP server set up...

File: hooks/push.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    $mail->setFrom(SMTP_SEND_FROM, $data->authorName);
+    $mail->addAddress(SMTP_SEND_TO);
+    $mail->addReplyTo(SMTP_SEND_BACK_TO);
```
Comment (nijel) [inline hooks/push.php]: Also please allow to set Approved header to custom value to send to moderated mailing list.

File: hooks/push.php (hunk 2 above)
Comment (williamdes) [reply hooks/push.php]: Yes, I will make the `SMTP_` optional.
phpmailer supports calling `mail()`

File: hooks/push.php (hunk 1 above)
Comment (williamdes) [reply hooks/push.php]: Yes, maybe, I find it strange that emails are send by your email and sent by mailing list but do not trigger SPF.
Maybe I did not understand everything with how works a mailing list.

Name | Value
-- | --
Message-ID | <5b55b6e498f08_b6c2ac949a46f7c42864 [EDITED] hookshot-fe-6b2eebc.cp1-iad.github.net.mail>
Date | Mon, 23 Jul 2018 04:07:16 -0700
From | Michal =?UTF-8?B?xIxpaGHFmQ==?= <michal [EDITED] cihar.com>
To | git [EDITED] phpmyadmin.net
Subject | [phpMyAdmin Git] [phpmyadmin/phpmyadmin] 7efcc1: Translated using  Weblate (Croatian)
Return-Path | <git-bounces [EDITED] phpmyadmin.net>

**Comment by williamdes** [comment]
@nijel I have done the changes you wanted, you only need to omit some `SMTP_` variables and it will work without a distant smtp server. You can add all the headers you want with the option `SMTP_HEADERS`.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**